core/optaplanner-constraint-streams-bavet/src/main/java/org/optaplanner/constraint/streams/bavet/common/index/NoneIndexerStorage.java

```diff
@@ -0,0 +1,86 @@
+package org.optaplanner.constraint.streams.bavet.common.index;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+
+/**
+ * Uses {@link LinkedHashMap} as tuple storage unless it is the first and only entry, in which case it uses fields.
+ * This helps avoid the overhead of creating and accessing a hash map if we only have 1 entry in the index.
+ * This situation occurs often enough that this optimization has been proven beneficial.
```

Comment:
Nitpick code style: avoid past tense in docs and javadocs: "has been proven beneficial" -> "is beneficial"